src/com/nkita/practice/collections/PersonStatisticHandler.java

```diff
@@ -176,26 +176,118 @@ public void checkFullStatisticWithoutDuplicates() {
         Set<Person> personSet = new LinkedHashSet<>(persons);
         System.out.println("Statistics with duplicate");
 
-        for (Person person : persons) {
-            personSet.add(person);
-            System.out.println(personSet);
-            return;
+        for (Person person : personSet) {
+            System.out.println(person);
         }
 
 
     }
 
 
-
     public void removeDuplicates() {
         Set<Person> personSet = new LinkedHashSet<>(persons);
         persons.clear();
         persons.addAll(personSet);
 
+    }
+
+
+    public void checkOldestPerson(Gender original) {
+
+        List<Person> female = new ArrayList<>();
+        List<Person> femaleAge = new ArrayList<>();
+        List<Person> male = new ArrayList<>();
+        List<Person> maleAge = new ArrayList<>();
+        List<Person> genderMaleAndFemale = new ArrayList<>();
+        List<Person> genderMaleAgeAndGenderFemaleAge = new ArrayList<>();
+
+
+        int max = 0;
+
+
+        for (Person person : persons) {
+
+            if (original == null) {
+                genderMaleAndFemale.add(person);
+            }
         }
 
+        for (Person person : genderMaleAndFemale) {
+            if (person.getAge() > max)
+                max = person.getAge();
+        }
+
+        for (Person person : genderMaleAndFemale) {
+            if (person.getAge() == max) {
+                genderMaleAgeAndGenderFemaleAge.add(person);
+            }
+
+        }
+
+
+        for (Person person : persons) {
+            if (person.getGender() == Gender.FEMALE) {
+                female.add(person);
+
+            }
+        }
+
+        for (Person person1 : female) {
+            if (person1.getAge() > max)
+                max = person1.getAge();
+        }
+
+        for (Person personAge : female) {
+            if (personAge.getAge() == max) {
+                femaleAge.add(personAge);
+            }
+
+        }
+
+
+        for (Person person : persons) {
+            if (person.getGender() == Gender.MALE) {
+                male.add(person);
+
+            }
+        }
+
+        for (Person person1 : male) {
+            if (person1.getAge() > max)
+                max = person1.getAge();
+        }
+
+        for (Person personAge : male) {
+            if (personAge.getAge() == max) {
+                maleAge.add(personAge);
+            }
+
+        }
```

Comment:
Lots of  "Foreach" cycles you can do all this stuff (check age and gender) in scope of one.